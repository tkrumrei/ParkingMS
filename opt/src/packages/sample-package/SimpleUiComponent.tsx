// SPDX-FileCopyrightText: 2023 Open Pioneer project (https://github.com/open-pioneer)
// SPDX-License-Identifier: Apache-2.0
import { FC } from "react";
import { useCommonComponentProps, CommonComponentProps } from "@open-pioneer/react-utils";
import { Box, chakra } from "@open-pioneer/chakra-integration";

export interface SimpleUiComponentProps extends CommonComponentProps {
    textToShow: string;
}

export const SimpleUiComponent: FC<SimpleUiComponentProps> = (props) => {
    const { textToShow } = props;
    const { containerProps } = useCommonComponentProps("simple-ui", props);
    return (
        <Box {...containerProps} border="1.5px solid" textAlign={"center"}>
            <chakra.p>{textToShow}</chakra.p>
        </Box>
    );
};
