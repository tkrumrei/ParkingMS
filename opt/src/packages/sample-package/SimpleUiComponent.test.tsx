// SPDX-FileCopyrightText: 2023 Open Pioneer project (https://github.com/open-pioneer)
// SPDX-License-Identifier: Apache-2.0
import { expect, it } from "vitest";
import { render, screen } from "@testing-library/react";
import { PackageContextProvider } from "@open-pioneer/test-utils/react";
import { SimpleUiComponent } from "./SimpleUiComponent";

it("simple ui component is rendered", async () => {
    /* The `PackageContextProvider` sets up  a test environment for chakra components and 
    enables hooks such as useService and useIntl within its children */
    render(
        <PackageContextProvider>
            <SimpleUiComponent
                textToShow="rendered successfully"
                data-testid="uiDiv"
            ></SimpleUiComponent>
        </PackageContextProvider>
    );

    const { element } = await waitForUi();
    expect(element.innerText).toBe("rendered successfully");
    expect(element).toMatchSnapshot();
});

async function waitForUi() {
    const uiDiv: HTMLDivElement | null = await screen.findByTestId<HTMLDivElement>("uiDiv");
    if (!uiDiv) {
        throw new Error("UI not rendered");
    }
    return {
        element: uiDiv
    };
}
