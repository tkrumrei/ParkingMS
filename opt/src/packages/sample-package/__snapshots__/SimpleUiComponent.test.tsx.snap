// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`simple ui component is rendered 1`] = `
<div
  class="simple-ui css-1uwnsjd"
  data-testid="uiDiv"
  data-theme="light"
>
  <p
    class="css-0"
    data-theme="light"
  >
    rendered successfully
  </p>
</div>
`;
